atmosphere/plugins/machine_validation.py

```diff
@@ -0,0 +1,140 @@
+        - Fail if image is a kernel or ramdisk
+        """
+        # Skip the machine if the owner was not the chromogenic image creator
+        # (tenant name must match admin tenant name)
+        # If that behavior changes, this snippet should be updated/removed.
+        if not cloud_machine:
+            return False
+        elif self._is_kernel_or_ramdisk(cloud_machine):
+            logger.info("Skipping cloud machine %s - kernel/ramdisk found" % cloud_machine)
+            return False
+        elif self._is_snapshot(cloud_machine):
+            logger.info("Skipping cloud machine %s - snapshot found" % cloud_machine)
+            return False
+        return True
+
+    def _contains_metadata(self, cloud_machine, metadata_key):
+        return True if hasattr(cloud_machine, metadata_key) else False
+
+    def _machine_authored_by_atmosphere(self, cloud_machine):
+        project_id = cloud_machine.get('owner')
+        owner_project = self.account_driver.get_project_by_id(project_id)
+        if not owner_project:
+            owner = cloud_machine.get('application_owner')
+            owner_project = self.account_driver.get_project(owner)
+        # Assumption: the atmosphere imaging author == the project_name set for the account_driver.
+        atmo_author_project_name = self.account_driver.project_name
+        if not owner_project:
+            logger.info(
+                "cloud machine %s - authored by project_id %s, not the Atmosphere author: %s",
+                cloud_machine.id, project_id, atmo_author_project_name)
+            return False
+        elif owner_project.name != atmo_author_project_name:
+            logger.info(
+                "cloud machine %s - authored by Tenant %s, not the Atmosphere author: %s",
+                cloud_machine.id, owner_project.name, atmo_author_project_name)
+            return False
+        return True
+
+    def _is_kernel_or_ramdisk(self, cloud_machine):
+        cloud_machine_name = cloud_machine.name if cloud_machine.name else ""
+        if any(cloud_machine_name.startswith(prefix) for prefix in ['eri-', 'eki-', 'ari-', 'aki-']):
+            return True
+        machine_type = cloud_machine.get('image_type', 'image')
+        if machine_type in ['ari', 'aki']:
+            return True
+        container_format = cloud_machine.get('container_format', '')
+        if container_format in ['ari', 'aki']:
+            return True
+        disk_format = cloud_machine.get('disk_format', '')
+        if disk_format in ['ari', 'aki']:
+            return True
+        return False
+
+    def _is_snapshot(self, cloud_machine):
+        cloud_machine_name = cloud_machine.name if cloud_machine.name else ""
+        if cloud_machine_name.startswith("ChromoSnapShot"):
+            return True
+        if cloud_machine.get('image_type', 'image') == 'snapshot':
+            return True
+        return False
+
+    def _machine_in_same_domain(self, cloud_machine):
```

Comment:
What is the need for this check?

Reply:
users with the Openstack 'admin' role see _all images, across all domains_

But that doesn't necessarily mean anyone on _the atmosphere-managed domain_ can launch the image. Best to skip it, in that case..

Reply:
roger!